File: 백준/Silver/9009. 피보나치/피보나치.py
N = int(input())

lst = []
for i in range(N):
    lst.append(int(input()))

for i in range(N):
    temp = lst[i]

    # 1) 사용할 피보나치 수 리스트를 만듦
    curIdx = 2
    curVal = 1
    fibonacci = [0, 1]
    while curVal <= temp:
        next = fibonacci[curIdx - 2] + fibonacci[curIdx - 1]
        if next > temp : break
        fibonacci.append(next)
        curVal = next
        curIdx += 1

    # 2) temp가 0이 될 때까지 피보나치 리스트에서 큰 수부터 추가
    tempLst = []
    while temp > 0 :
        last = fibonacci.pop()
        if last <= temp : 
            tempLst.append(last)
            temp -= last
    
    tempLst.sort()
    print(*tempLst)

File: 백준/Silver/9009. 피보나치/피보나치.js
const INPUT = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")
  // .split(' ')
  .map(Number);

function solution() {
  const N = Number(INPUT[0]);
  const arr = INPUT.slice(1).map(Number);
  let answer = "";

  // 1)
  for(let i = 0; i < N; i++){
    let temp = arr[i];
    
    let curIdx = 2;
    let curVal = 1;
    const fibonacci = [0, 1];

    while(curVal <= temp){
      const next = fibonacci[curIdx - 2] + fibonacci[curIdx - 1];
      
      if(next > temp) break;

      fibonacci.push(next);
      curVal = next;
      curIdx++;
    }

    // 2)
    const tempLst = [];
    while(temp > 0){
      const last = fibonacci.pop();
      if(last <= temp){
        tempLst.push(last);
        temp -= last;
      }
    }

    tempLst.sort((a, b) => a - b);

    answer += `${tempLst.join(" ")}\n`
  }
  console.log(answer);
}

solution();